Sources/MQTTNIO/ChannelHandlers/MQTTChannelHandler.swift

```diff
@@ -318,6 +326,36 @@ class MQTTChannelHandler: ChannelDuplexHandler {
     // MARK: - Pingreq Handling
 
     func updatePingreqTimeout(_ timeout: TimeAmount) {
-        self.pingreqHandler?.updateTimeout(timeout)
+        self.pingreqTimeout = timeout
+    }
+
+    func schedulePingreqTask(_ context: ChannelHandlerContext) {
+        guard context.channel.isActive else { return }
+
+        self.pingreqTask = context.eventLoop.scheduleTask(deadline: self.lastPingreqEventTime + self.pingreqTimeout) {
```

Comment:
Can we use `scheduleCallback` instead. It is a cleaner API and should reduce the number of allocations